Corona-Warn-App/src/main/java/de/rki/coronawarnapp/service/submission/SubmissionService.kt

```diff
@@ -73,12 +73,20 @@ object SubmissionService {
     /**
      * extracts the GUID from [scanResult]. Returns null if it does not match the required pattern
      */
+    @SuppressWarnings("ReturnCount")
     fun extractGUID(scanResult: String): String? {
-        val potentialGUID = scanResult.substringAfterLast("?", "")
-        return if (potentialGUID.isEmpty())
-            null
-        else
-            potentialGUID
+        if (scanResult.length > SubmissionConstants.MAX_QR_CODE_LENGTH)
+            return null
+
+        if (scanResult.count { it == SubmissionConstants.GUID_SEPARATOR } != 1)
+            return null
+
+        val potentialGUID = scanResult.substringAfterLast(SubmissionConstants.GUID_SEPARATOR, "")
+
+        if (potentialGUID.isEmpty() || potentialGUID.length > SubmissionConstants.MAX_GUID_LENGTH)
+            return null
+
+        return potentialGUID
```

Comment:
This is not a good implementation imo. It uses null return, has too many return statements and is not described enough. I would ask to discuss this again or change the implementation here.

Reply:
@jakobmoellersap thanks for the feedback! `null` signals that the pattern is not valid. Would you prefer splitting it into something like `isValidGUID` returning Boolean and extractGUID returning String?

IMO the multiple returns make it easy to read as each if covers one constraint. Of course I can merge some of them to reduce the number of returns but I´m not sure if this adds benefit

Reply:
So I think I would create a return value that is able to only return if all conditions are not met. Furthermore you use string operations, so you could make use of lambdas like filtering on the string or the length. That way I think we will be able to reduce the return amounts :)

Regarding Null Return: Either you throw an Exception with a catch around the call or you leave it as is. But first lets try to reduce the return amounts. 👍